### PR TITLE
Skip special files to avoid NPR in RockerUtil.listFileTree()

### PR DESCRIPTION
**Pull request by lenaschimmel**
This should fix fizzed#159.

### CODE REVIEW

**Comment by lenaschimmel** [comment]
After I submitted the PR, I was still unsure if this is a correct and complete solution, or if there are cases of non-normal files which should be handled explicitly. I'm writing this down so that whoever reviews the PR can retrace my reasoning.

As far as I can tell, these are the special cases that may exist:

1. process intercom
1.1. FIFO pipe / named pipe
1.2. UNIX domain socket
2. device
2.1. block device 
2.2. character device
3. symbolic link
3.1. broken symbolic link (target does not exist)
3.2. working symbolic link (target exists and might be a regular file, a regular directory, or any of these special cases)
4. hard link

For cases 1 and 2, I can't see any proper way to handle them except ignoring, and so I can't see why any user would want to have them handled. I *think* they might also cause a crash prior to the PR, and would be ignored when merging the PR.

Case 3: working symbolic links, should behave like regular files or directories, no matter if the PR is merged or not. For broken symbolic links, it's a matter of crashing vs. ignoring.

And case 4, hard links, can only be files, not directories, and are already handled correctly prior to the PR and will not be affected by it.

Bottom line: I believe this PR handles multiple possible error conditions and does not introduce any new problems.

**Comment by jjlauer** [comment]
Cleaning up old PRs for Java 21.  I'm not sure this is relevant (how many people have unix pipes, etc. in their rocker template tree?)  Feel free to reopen if you still need it.
